### PR TITLE
Migrate poetry to hatch

### PR DESCRIPTION
**Pull request by koxudaxi**
WIP

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,8 +1,9 @@
 [tool.poetry]
```
Comment (pauleveritt) [inline pyproject.toml:1]: This should be [project] not [tool.poetry]

File: pyproject.toml [hunk 2]
```diff
@@ -13,45 +14,45 @@ classifiers = [
...
 
+[tool.hatch.version]
+path = "src/viewdom/version.py"
+
+[tool.hatch.envs.test]
+dependencies = [
+    "pytest~=7.4.3",
+    "sybil~=3.0.1",
+    "coverage[toml]~=6.3",
+    "xdoctest[colors]~=0.15.10",
```
Comment (pauleveritt) [inline pyproject.toml:38]: xdoctest won't install without distutils (meaning, Python 3.12) and I believe we are dropping this anyway.

File: pyproject.toml [hunk 3]
```diff
@@ -13,45 +14,45 @@ classifiers = [
...
-nox-poetry = "^1.0.3"
-furo = "^2022.3.4"
-myst-parser = "^0.17.0"
-sybil = "^3.0.1"
 
+[tool.hatch.version]
+path = "src/viewdom/version.py"
+
+[tool.hatch.envs.test]
+dependencies = [
```
Comment (pauleveritt) [inline pyproject.toml:34]: Can we try adding path = ".venv" to try and support local virtual environments for PyCharm?

Also, let's edit the CONTRIBUTING.md to show how to generate a venv for PyCharm: $  hatch run +py=3.12 test:run

**Review by pauleveritt — CHANGES_REQUESTED** [review]

**Comment by koxudaxi** [comment]
@pauleveritt 
Could you please release the new version of hopscotch?
hatch can't resolve dependencies
```
❯ hatch run test                
ERROR: Ignored the following versions that require a different python version: 0.3.1 Requires-Python >=3.11,<4.0
ERROR: Could not find a version that satisfies the requirement hopscotch>=0.3.1 (from versions: 0.1.2, 0.1.3, 0.1.4, 0.1.5, 0.1.6, 0.2.0, 0.2.1, 0.3.0)
ERROR: No matching distribution found for hopscotch>=0.3.1
(base) 
````

File: pyproject.toml (hunk 2 above)
Comment (koxudaxi) [reply pyproject.toml:38]: I just removed

File: pyproject.toml (hunk 1 above)
Comment (koxudaxi) [reply pyproject.toml:1]: I just fixed it.

**Comment by koxudaxi** [comment]
@pauleveritt 
It's happen again.
the `hopscotch==0.3.1`'s requirement of version is borken?
```
hatch shell
ERROR: Ignored the following versions that require a different python version: 0.3.1 Requires-Python >=3.11,<4.0
ERROR: Could not find a version that satisfies the requirement hopscotch>=0.3.1 (from viewdom) (from versions: 0.1.2, 0.1.3, 0.1.4, 0.1.5, 0.1.6, 0.2.0, 0.2.1, 0.3.0)
ERROR: No matching distribution found for hopscotch>=0.3.1
(base) 

```

File: pyproject.toml (hunk 3 above)
Comment (koxudaxi) [reply pyproject.toml:34]: I have updated the CONTRIBUTING.md

**Comment by koxudaxi** [comment]
@pauleveritt 
Can I get the permission to run the CI?

**Comment by pauleveritt** [comment]
Sorry, I should have added you as a contributor, I just did so.

—Paul

**Review by pauleveritt — APPROVED** [review]

**Pull request merged**